superdesk/io/format_document_for_preview.py

```diff
@@ -0,0 +1,43 @@
+from flask import Blueprint, request, Response
+import superdesk
+from superdesk import get_resource_service
+from superdesk.publish.formatters import get_formatter
+from apps.content_types import apply_schema
+
+
+bp = Blueprint('format_document', __name__)
+
+
+def get_mime_type(formatter_qcode):
+    if(formatter_qcode == 'newsmlg2'):
+        return 'text/xml'
+    else:
+        return 'application/json'
+
+
+@bp.route('/format-document-for-preview/', methods=['GET', 'OPTIONS'])
```

Comment:
imo it should be using POST

Reply:
Why? Something is created on your side with it? What if we want to get it multiple times? I think that GET would be better here.

Reply:
client sends a payload and gets it processed, imo that's what post is for, with get and sending only id it could get cached

Reply:
POST should be used when: 
``` 
* Posting a message to a bulletin board, newsgroup, mailing list,
      blog, or similar group of articles;

* Creating a new resource that has yet to be identified by the
      origin server; and

* Appending data to a resource's existing representation(s).
```

So it depends if this endpoint creates some resource at backend. If it just process it every time based on provided parameters then GET will be better.

Reply:
If I send a POST request twice with this same data then I should get another item in response or information about duplicated resource created. If it gives me this same (identical) response every time then it's GET.

Reply:
imo it's more like:
```
- Providing a block of data, such as the result of submitting a
        form, to a data-handling process;
```
from https://www.w3.org/Protocols/rfc2616/rfc2616-sec9.html

it depends if client will send just id or the full item, I think it should send the full item so you can get preview before saving the changes. and then it should be POST. if it's only id GET is fine

Reply:
As we are using the ID as explained https://github.com/superdesk/superdesk-core/pull/1316/files#r203367817 let's go with `GET`